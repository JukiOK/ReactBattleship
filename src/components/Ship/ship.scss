@import '../../constants/colors';


.ship-container {
}
.ship {
  width: 39px;
  height: 38px;
}
