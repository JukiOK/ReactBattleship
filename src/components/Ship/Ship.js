import {useState, useEffect} from 'react';
import colors from '../../constants/colors.js';
import { useDrag } from 'react-dnd';
import { ItemTypes } from '../../constants/ItemTypes';

require('./ship.scss');

function Ship(props) {
  const [{isDragging}, drag] = useDrag({
      item: {
        type: ItemTypes.SHIP,
      },
      collect: monitor => ({
        isDragging: !!monitor.isDragging()
      })
    })

  function ship() {
    let color;
    switch (props.type) {
      case 'boat':
        color = colors.yellow;
        return (
          <div className="ship-container" data-testid="ship-boat">
            <div className="ship" style={{backgroundColor: color}}>
            </div>
          </div>
        )

        break;
      case 'destroyer':
        color = colors.green;
        return (
          <div className="ship-container">
            {
              [1, 2].map((index) => (
                <div className="ship" style={{backgroundColor: color}} key={index}>
                </div>
              ))
            }
          </div>
        )
        break;
      case 'submarine':
        color = colors.blue;
        return (
          <div className="ship-container">
            {
              [1, 2, 3].map((index) => (
                <div className="ship" style={{backgroundColor: color}} key={index}>
                </div>
              ))
            }
          </div>
        )
        break;
      case 'carrier':
        color = colors.purple;
        return (
          <div className="ship-container">
            {
              [1, 2, 3, 4].map((index) => (
                <div className="ship" style={{backgroundColor: color}} key={index}>
                </div>
              ))
            }
          </div>
        )
        break;
      default:
        break;
    }
  }
  return (
    <div ref={drag}
      style={{opacity: isDragging ? 0.5 : 1,}}
    >
      {ship()}
    </div>
  )
}

export default Ship;
