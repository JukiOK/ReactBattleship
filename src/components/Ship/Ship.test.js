import React from 'react';
import {render, unmountComponentAtNode } from 'react-dom';
import Ship from './Ship';
import { act } from "react-dom/test-utils";
import renderer from 'react-test-renderer';

let div;

beforeEach(() => {
  div = document.createElement('div');
  document.body.appendChild(div);
})

afterEach(() => {
  unmountComponentAtNode(div);
  div.remove();
  div = null;
})

it("renders without crash", () => {
  render(<Ship></Ship>, div);
})

it("render ship", () => {

})
//
// it("match snapshot", () => {
//   const tree = renderer.create(<Ship type={'boat'} />).toJSON();
//   expect(tree).toMatchSnapshot();
// })
