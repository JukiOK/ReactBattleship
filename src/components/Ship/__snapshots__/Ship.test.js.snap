// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div>
  <div
    className="ship-container"
    data-testid="ship-boat"
  >
    <div
      className="ship"
      style={
        Object {
          "backgroundColor": "#FFD046",
        }
      }
    >
      #FFD046
    </div>
  </div>
</div>
`;
